### PR TITLE
Delay importing deepspeed comm due for perf

### PR DESCRIPTION
**Pull request by jiminha**
# What does this PR do?
We are seeing 2-3 secs total run time slowness with most of the mpi multicard tests with deepspeed installed compare to the one without.  We found this part is being called for every single model run which causing the slowdown. 
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/models/mixtral/modeling_mixtral.py#L65

### CODE REVIEW

**Comment by jiminha** [comment]
@jychen-habana  could you review this change, and also run the test which will be impacted with this change?  I was trying to run but getting some error saying npz file is missing(required for quantization). Please validate if deepspeed part is working ok. 

Also, is this mixtral model always use deepspeed for multi-card run?

File: optimum/habana/transformers/models/mixtral/modeling_mixtral.py [hunk 1]
```diff
@@ -294,10 +289,14 @@ def gaudi_mixtral_block_sparse_moe_forward(self, hidden_states: torch.Tensor) ->
...
 
-    if dist and dist.is_initialized():
+    if is_deepspeed_available():
+        from deepspeed import comm as dist
+
         output_tensors = [router_logits.clone() for _ in range(dist.get_world_size())]
         dist.all_gather(output_tensors, router_logits)
         router_logits = torch.cat(output_tensors, dim=1)
+    else:
+        print("Not using HPU DeepSpeed.")
```
Comment (regisss) [inline optimum/habana/transformers/models/mixtral/modeling_mixtral.py:299]: Let's remove that actually, there will already be a similar log before if DeepSpeed is used.

**Review by regisss — APPROVED** [review]
LGTM, good catch!

Mixtral does not necessarily need DeepSpeed to run properly. It can run on one Gaudi2 card in bf16. I'll wait for @jychen-habana if there are additional tests to carry out.

**Comment by jychen-habana** [comment]
@jiminha @regisss 

Looks good to me. Added a little change for the case that deepspeed installed but not used.

For FP8 quantization, the error "npz file missing" means that in quantization mode, it can not find the measured bf16 data in "./hqt_output". So, for single-card quantization just use single-card measurement.

For Mixtral inference multi-card, yes, it's deepspeed TP.

**Review by jychen-habana — APPROVED** [review]
LGTM

**Comment by SushantGautam** [comment]
Traceback (most recent call last):
  File "/optimum-habana/examples/text-generation/run_generation.py", line 626, in <module>
    main()
  File "/optimum-habana/examples/text-generation/run_generation.py", line 278, in main
    model, tokenizer, generation_config = initialize_model(args, logger)
  File "/optimum-habana/examples/text-generation/utils.py", line 379, in initialize_model
    setup_model(args, model_dtype, model_kwargs, logger)
  File "/optimum-habana/examples/text-generation/utils.py", line 168, in setup_model
    habana_quantization_toolkit.prep_model(model)
  File "/usr/local/lib/python3.10/dist-packages/habana_quantization_toolkit/prepare_quant/prepare_model.py", line 12, in prep_model
    prepare_model(model)  # registers hooks
  File "/usr/local/lib/python3.10/dist-packages/habana_quantization_toolkit/_hook_method/__init__.py", line 45, in prepare_model
    return quantize_hooks(model, mod_list)
  File "/usr/local/lib/python3.10/dist-packages/habana_quantization_toolkit/_hook_method/quantize.py", line 63, in quantize_hooks
    measurement=load_measurements(config['measure_file'])
  File "/usr/local/lib/python3.10/dist-packages/habana_quantization_toolkit/_hook_method/measure.py", line 125, in load_measurements
    d = load_file(fname_np, np.ndarray, fail_on_file_not_exist=config['scale_method'] not in [ScaleMethod.WITHOUT_SCALE, ScaleMethod.UNIT_SCALE])
  File "/usr/local/lib/python3.10/dist-packages/habana_quantization_toolkit/_hook_method/common.py", line 109, in load_file
    raise FileNotFoundError(f"Failed to load file {fname}")
FileNotFoundError: Failed to load file ./hqt_output/measure_hooks_maxabs.npz

**Pull request merged**